augur/io/file.py

```diff
@@ -22,8 +24,12 @@ def open_file(path_or_buffer, mode="r", **kwargs):
         File handle object
 
     """
-    try:
+    if isinstance(path_or_buffer, (str, os.PathLike)):
         with xopen(path_or_buffer, mode, **kwargs) as handle:
             yield handle
-    except TypeError:
+
+    elif isinstance(path_or_buffer, (IOBase, PipedCompressionReader, PipedCompressionWriter)):
         yield path_or_buffer
+
+    else:
+        raise NotImplementedError(f"Type {type(path_or_buffer)} is not supported.")
```

Comment:
_non-blocking_

What's the thought behind using `NotImplementedError` vs `TypeError`?

Reply:
Admittedly not much thought – I was thinking this might support more types in the future. After adding more thought now, I can't come up with any examples and none are documented, so I'll change it to `TypeError`.

Reply:
Done in [force-push](https://github.com/nextstrain/augur/compare/205b70e03a8099a383d05addb2024a0b6e14e948..2d02ec2bea8beaf05ab146da179ff6411b131217).